adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java

```diff
@@ -414,6 +431,66 @@ public DisableRequest createDisableRequest(final String merchantAccount, final S
         return abstractPaymentRequest;
     }
 
+    public TerminalAPIRequest createTerminalAPIRequest(final CartData cartData, CustomerModel customer, RecurringContractMode recurringContractMode) throws Exception {
+        SaleToPOIRequest saleToPOIRequest = new SaleToPOIRequest();
+
+        //TODO get Service ID generated from session
+        long id = Calendar.getInstance().getTimeInMillis() % 10000000000L;
+
+        MessageHeader messageHeader = new MessageHeader();
+        messageHeader.setProtocolVersion("3.0");
```

Comment:
Do we need to hardocode this ?

Reply:
This is the version of Terminal API. The latest one is 3.0.